app/assets/javascripts/spree/frontend/paypal_button.js

```diff
@@ -19,7 +19,7 @@ $(document).ready(function() {
       }).
       load(function() {
         var paypalOptions = {
-          flow: 'vault',
+          flow: window.payPalFlow,
```

Comment:
This button is rendered in the cart. Are we sure that the global variable is set?

Reply:
@tvdeyen I agree that this is a bit brittle. Do you have any suggestions?

Reply:
Not without refactoring how we setup the paypal button in the cart. Ideally we would do it like we do it in checkout payment. Loading the script and setting in up in a view partial.

Reply:
@tvdeyen so something like this, right?

```erb
<%= render 'paypal_button' %>
```

And `_paypal_button.html.erb` would be just:

```erb
<script>
$(document).ready(function() {
  if (document.getElementById("empty-cart")) {
    $.when(
      $.getScript("https://js.braintreegateway.com/web/3.22.1/js/client.min.js"),
      $.getScript("https://js.braintreegateway.com/web/3.22.1/js/paypal.min.js"),
      $.getScript("https://js.braintreegateway.com/web/3.22.1/js/data-collector.min.js")
    ).done(function() {
      $('<script/>').attr({
        'data-merchant' : "braintree",
        'data-id' : "paypal-button",
        'data-button' : "checkout",
        'data-color' : "blue",
        'data-size' : "medium",
        'data-shape' : "pill",
        'data-button_type' : "button",
        'data-button_disabled' : "true"
      }).
      load(function() {
        var paypalOptions = {
          flow: '<%= SolidusPaypalBraintree::Gateway.first.preferred_paypal_flow %>',
          enableShippingAddress: true
        }
        var button = new SolidusPaypalBraintree.createPaypalButton(document.querySelector("#paypal-button"), paypalOptions);
        return button.initialize();
      }).
      insertAfter("#content").
      attr('src', 'https://www.paypalobjects.com/api/button.js?')
    });
  }
});
</script>
```
I can go ahead and implement it, but I'm afraid this might be a breaking change? I guess we _could_ keep the original asset for backwards compatibility and implement the partial for when you want to tweak the flow, but I'm not sure this is a good idea. What do you think?

Reply:
Thanks, yes. Also taken care of in #170